src/main/java/org/opensearch/knn/index/JNIService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index;
+
+import org.opensearch.knn.common.KNNConstants;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Map;
+
+/**
+ * Service to interact with plugin's jni layer. Class dependencies should be minimal
+ *
+ * In order to compile C++ header file, run:
```

Comment:
i remember that we have to comment KNNConstants to generate. is that still the case?

Reply:
No need to comment out. It will work as is.